evaluator.activitydefinition/src/main/java/org/opencds/cqf/cql/evaluator/activitydefinition/BaseActivityDefinitionProcessor.java

```diff
@@ -63,16 +67,67 @@ public IBaseResource apply(IIdType theId, String subjectId, String encounterId,
       String practitionerId, String organizationId, String userType, String userLanguage,
       String userTaskContext, String setting, String settingContext, IBaseParameters parameters,
       LibraryEngine libraryEngine) {
-    this.subjectId = subjectId;
-    this.parameters = parameters;
-    this.libraryEngine = libraryEngine;
     T activityDefinition = (T) this.repository.read(ActivityDefinition.class, theId);
     if (activityDefinition == null) {
       throw new IllegalArgumentException("Couldn't find ActivityDefinition " + theId);
     }
+    return apply(activityDefinition, subjectId, encounterId, practitionerId, organizationId,
+        userType, userLanguage, userTaskContext, setting, settingContext, parameters, libraryEngine);
+  }
+
+  public IBaseResource apply(String url, String subjectId, String encounterId,
```

Comment:
@JPercival  What is the good type to use for Url?

The `IPrimitiveType<String>` collides with `IIdType` and actual `CanonicalType`/`UrlType` is version specific